src/lib/Surveys/surveyUtils.js

```diff
@@ -0,0 +1,121 @@
+import { showSurveyModal, surveyStore } from '$stores/surveyStore.js';
+
+export async function loadSurveys(stop = null, userId = null) {
+	try {
+		const response = await fetch(`/api/oba/surveys?userId=${userId}`);
+		if (!response.ok) throw new Error('Failed to fetch surveys');
+
+		const data = await response.json();
+		const validSurveys = getValidSurveys(data.surveys);
+
+		let selectedSurvey = null;
+
+		if (stop) {
+			selectedSurvey =
+				getValidStopSurvey(validSurveys, stop) || getShowSurveyOnAllStops(validSurveys);
+		} else {
+			selectedSurvey = getMapSurvey(validSurveys);
+		}
+
+		surveyStore.set(selectedSurvey);
+
+		showSurveyModal.set(selectedSurvey?.show_on_map === true);
+	} catch (error) {
+		console.error('Error loading surveys:', error);
+	}
+}
+
+export function getValidSurveys(surveys) {
+	const now = new Date();
+	return surveys.filter(
+		(survey) =>
+			new Date(survey.end_date) > now &&
```

Comment:
it's valid for a survey to not have an end date, which makes it not appear on-screen as this filters it out as being invalid.